Sources/LLVM/JIT.swift

```diff
@@ -0,0 +1,112 @@
+import cllvm
+
+/// JITError represents the different kinds of errors the JIT compiler can throw
+public enum JITError: Error, CustomStringConvertible {
+    /// The JIT was unable to be initialized. A message is provided explaining
+    /// the failure.
+    case couldNotInitialize(String)
+
+    /// A human-readable description of the error.
+    public var description: String {
+        switch self {
+        case .couldNotInitialize(let message):
+            return "could not initialize JIT: \(message)"
+        }
+    }
+}
+
+/// A JIT is a Just-In-Time compiler that will compile and execute LLVM IR that
+/// has been generated in a `Module`. It can execute arbitrary functions and
+/// return the value the function generated, allowing you to write interactive
+/// programs that will run as soon as they are compiled.
+class JIT {
+    /// The underlying LLVMExecutionEngineRef backing this JIT
+    internal let llvm: LLVMExecutionEngineRef
+
+    /// Creates a Just In Time compiler that will compile the code in the
+    /// provided `Module` to the architecture of the provided `TargetMachine`,
+    /// and execute it.
+    ///
+    /// - parameters:
+    ///   - module: The module containing code you wish to execute
+    ///   - machine: The target machine which you're compiling for
+    /// - throws: JITError
+    init(module: Module, machine: TargetMachine) throws {
+        var jit: LLVMExecutionEngineRef?
+        var error: UnsafeMutablePointer<Int8>?
+        if LLVMCreateExecutionEngineForModule(&jit, module.llvm, &error) != 0 {
+            let str = String(cString: error!)
+            throw JITError.couldNotInitialize(str)
+        }
+        guard let _jit = jit else {
+            throw JITError.couldNotInitialize("JIT was NULL")
+        }
+        self.llvm = _jit
+    }
+
+
+    /// Runs the specified function with the provided arguments by compiling
+    /// it to machine code for your specific architecture.
```

Comment:
Nit: `this JIT's  specified architecture` or `the target architecture used to initialize this JIT`.